public/static/docs/api-reference/open.md

```diff
@@ -0,0 +1,139 @@
+only be used as a _context manager_ (using the `with` keyword as shown in the
+examples below). There is no `close()`.
+
+💡 Note that `dvc.api.open` is able to
+[stream](https://docs.python.org/3/library/io.html) the file directly from
+**most**
+[remote types](/doc/command-reference/remote/add#supported-storage-types)
+(local, S3, Azure, GCP, OSS, SSH, HDFS). Otherwise, the file is downloaded
+regularly into a temporary local path before the file object is made available.
+
+> This has similar uses as the `dvc get` and `dvc import` CLI commands.
+
+## Parameters
+
+- **`path`** - used to specify the location of the target artifact within the
+  source project in `repo`, relative to the project's root.
+
+- `repo` - specifies the location of the source DVC project. Both HTTP and SSH
+  protocols are supported for online Git repository URLs (e.g.
+  `[user@]server:project.git`). `repo` can also be a local file system path to
+  an "offline" project. If not supplied, this defaults to the current working
+  directory.
+
+  > A `dvc.exceptions.NotDvcRepoError` is thrown if `repo` is not a valid DVC
+  > project.
+
+- `rev` - (optional)
+  [Git-revision](https://git-scm.com/book/en/v2/Git-Internals-Git-References)
+  (such as a branch name, a tag, or a commit hash). `rev` only has an effect
+  when a URL is supplied as parameter to `repo`. If not supplied, it uses the
+  default Git revision, `HEAD`.
+
+- `remote` - (optional) name of the [DVC remote](/doc/command-reference/remote)
+  to fetch the target artifact from. If not supplied, the default depends on the
+  value of `repo`. The local cache is used when `repo` is the current working
+  directory (default value of `repo`). when `repo` is an external repository
+  URL, the default project remote is used.
+
+  > A `dvc.exceptions.NoRemoteError` is thrown if no `remote` is specified and
+  > the project has no default remote.
+
+- `mode` - (optional) mirrors the namesake parameter in builtin
+  [`open()`](https://docs.python.org/3/library/functions.html#open). Defaults to
+  `"r"` (read).
+
+- `encoding` - (optional) used to decode contents to a string. Mirrors the
+  namesake parameter in builtin `open()`. Defaults to `"utf-8"`.
+
+## Example: process XML file from an external DVC repository
+
+```py
+from xml.dom.minidom import parse
+
+import dvc.api
+
+with dvc.api.open(
+  "get-started/data.xml",
+  "https://github.com/iterative/dataset-registry"
+) as fd:
+  xmldom = parse(fd)
+  # ... Process elements
+```
+
+> See also `dvc.api.read` for a more direct way to read the complete contents of
+> a file <abbr>artifact</abbr>.
+
+## Example: use a file from the local cache
+
+In this case we don't supply a `repo` value, which means the current working
+directory will be tried instead, so make sure that the code is run from within a
+<abbr>DVC project</abbr>:
+
+```py
+import dvc.api
+
+with dvc.api.open('data/nlp/words.txt') as fd:
+  print(fd.name)
```

Comment:
Opening a file to find its cache is quite a hack. A more real code sample might be:
```python
with dvc.api.open('data/nlp/words.txt') as fd:
    for word in fd:
        # ... process a word
```
Also it's standard for python to use 4 spaces.

Reply:
Also, you can use this example to showcase an encoding param:
```python
with dvc.api.open('data/nlp/words.txt', encoding="utf-8") as fd:
    # ...
```

Reply:
> A more real code sample...

It was just the first thing I thought of, using your example now.

> it's standard for python to use 4 spaces

OK. Reformatted per https://www.python.org/dev/peps/pep-0008/#indentation please check because the multi-line `with` statements are tricky... And we only have 72 characters or so, otherwise the code samples start to show a horizontal scrolling bar.

Reply:
> use this example to showcase an encoding

Good idea, added to the same example. Thanks!

Reply:
@Suor this is the only possibly pending comment from your review here I think. Can you review the examples again? Thanks